### PR TITLE
IRO-1399 - Leaderboard Pagination

### PR DESCRIPTION
**Pull request by brekk**
## Summary

Make `/leaderboard` data paginated.

## Testing Plan

👀 

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] Nope
```

### CODE REVIEW

File: pages/leaderboard.tsx [hunk 1]
```diff
@@ -64,6 +65,8 @@ const FIELDS = {
 }
 const CTA = `Our incentivized testnet leaderboard shows you who the top point getters are for all-time score, miners, bug catchers, net promoters, node hosting, and more! Click someone’s user name to see a breakdown of their activity. Points are earned during weekly cycles which begin Monday, 12:00am UTC and end Sunday 11:59pm UTC.`
 
+const PAGINATION_LIMIT = 3
```
Comment (brekk) [inline pages/leaderboard.tsx:68]: This is set temporarily, it needs to be changed to 25 _before_ this merges.

**Comment by brekk** [comment]
Sounds like there's a bug on the API side regarding `before` sorting, keeping this in Draft but it's ready for review.

File: apiClient/client.ts [hunk 2]
```diff
@@ -72,15 +72,21 @@ export async function updateUser(id: number, partial: PartialUser) {
...
+  before,
 }: {
   search?: string
   country_code?: string
   event_type?: string
-} = {}): Promise<ListLeaderboardResponse | ApiError> {
+  limit?: number
+  after?: number
+  before?: number
+}): Promise<ListLeaderboardResponse | ApiError> {
```
Comment (dguenther) [inline apiClient/client.ts:89]: It'd be nice to rename the response type to `ListUsersResponse` if we're going to rename the function (I agree renaming the function makes sense)

File: pages/users/[id].tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ import Footer from 'components/Footer'
 import Navbar from 'components/Navbar'
 import Loader from 'components/Loader'
 import { Container as OffsetBorderContainer } from 'components/OffsetBorder'
-import EventsPaginationButton from 'components/user/EventsPaginationButton'
+import EventsPaginationButton from 'components/PaginationButton'
```
Comment (dguenther) [inline pages/users/[id].tsx:12]: This could be renamed to `PaginationButton` like on the leaderboard page, to match up the names.

**Review by dguenther — APPROVED** [review]
Sounds like we'll wait to ship to master until the `before` issue with `order_by=rank` is fixed on the API, but otherwise looks good!

**Pull request merged**